### ISSUE TITLE
Babun setup on HyperTerm - Unable to find proper settings

### ISSUE THREAD

**Issue by labnux** [zeit/hyper#1398]
- [X] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [X] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Windows 10 Pro
- **Hyper.app version**: Hyper 1.1.0 (installed through Chocolatey)
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: https://gist.github.com/labnux/97df2ecb95a010ad6aff8bb181ecc676
- **Relevent information from devtools** _(CMD+SHIFT+I on Mac OS, CTRL+SHIFT+I elsewhere)_: N/A
- **The issue is reproducible in vanilla Hyper.app**: Is-Vanilla (normally)

## Issue
Hello,

I've been wanting to try out Hyper.app for quite some time now. However, as a system administrator working with Windows (not really but choice), I need a fully functioning environment. After trying to configure - in vein - the Cmder bash.exe within Hyper.app, I'm currently trying to fit Babun in, which seems pretty awesome considering all the features it offers. Problem is: I'm unable to launch it properly.

On Babun FAQ, they say to enter the following configuration in ConEmu (which works BTW):
Task parameters - /icon "%userprofile%\.babun\cygwin\bin\mintty.exe" /dir "%userprofile%"
Commands %userprofile%\.babun\cygwin\bin\mintty.exe /bin/env CHERE_INVOKING=1 /bin/zsh.exe

So, I tried to configure that in Hyper.app format and wasn't able to do much. Anyone would have an idea? I've been given some error messages saying that /bin/env doesn't exist (No such file or directory).

Basically, Babun installs itself in %USERDIR%\.babun with all its dependencies (Cygwin, modules, etc). My impression is with ConEmu, the arguments are sent directly to the terminal environment, which explains why /bin/env and /bin/bash.exe works.

Is there proper documentation about the shell, shellArgs and env variables in .hyper.js configuration file? Any idea on how to proceed in this case?

Thank you in advance.

Richard.

**Comment by schmic** [comment]
I'm fiddling with Hyper as well and here is what I'm using with Babun:

```
    // the shell to run when spawning a new session (i.e. /usr/local/bin/fish)
    // if left empty, your system's login shell will be used by default
    shell: 'C:\\Users\\schmic\\.babun\\cygwin\\bin\\zsh.exe',

    // for setting shell arguments (i.e. for using interactive shellArgs: ['-i'])
    // by default ['--login'] will be used
    shellArgs: ['--login'],

    // for environment variables
    env: {
      CHERE_INVOKING: 1
    },
```

I do get a broken commandline when using backspace though, some characters get removed but are still displayed. Haven't looked into that more closely though.

**Comment by labnux** [comment]
I'll certainly give it a try. Thanks for the tip!

**Comment by labnux** [comment]
Got the same behavior as @schmic with these settings. Backspace key will softlock the terminal, also get error messages regarding the version it's getting.

**Comment by Hammster** [comment]
Babun did start with @schmic config, i can also confirm that the command line is getting messed up if you try to delete via backspace. It is not bound to certain characters or the speed the characters are typed or deleted, also changing the zsh theme does not change this behavior.

When running nano, the buffer still contains the commandline output. so i guess something is off with the screen clearing before the buffer is refilled.

**Comment by martin-wiseweb** [comment]
I _love_ the look of Hyper but being stuck on Windows at work makes it difficult to find this a good terminal for anything other than file watchers, etc. To do actual work, I'm still using Cmder until I can get Linux commands emulation working on Hyper - either via Babun, Cmder, whatever

**Comment by Fed03** [comment]
👍

**Comment by schmic** [comment]
This issue seems to be fixed now with the settings I posted above.
Also works fine with Msys2 and zsh now. Cheers for v1.2.1.

**Comment by Hammster** [comment]
Something is still off for me, I updated to v1.2.1 deleted my old config. And changes the value according to the post of @schmic. 

The issues I described are still there.

testcase scenario:
```
ls -la
nano
```

Results in this 
![untitled](https://cloud.githubusercontent.com/assets/1093709/23065680/1917a18e-f517-11e6-9cd8-442ba44d7ee1.jpg)

**Comment by schmic** [comment]
@Hammster Correct, vim is still broken for me as well.

The "broken commandline" with characters remaining there while using backspace is definetly gone for me.

**Comment by Hammster** [comment]
@schmic they are also gone for me, I should have made clear which problem still exist 🦊

**Comment by meeech** [comment]
can confirm works fine for me babun, no weird issues using backspace. But does anyone have any ideas about the extra % on every line? getting that too - 
![image](https://cloud.githubusercontent.com/assets/4623/23195232/2cf46270-f880-11e6-95cb-2726c389284b.png)

**Comment by schmic** [comment]
@meeech Put `unset PROMPT_SP` in your .zshrc

**Comment by daagar** [comment]
For anyone mildly puzzled, it is actually `unsetopt PROMPT_SP` instead of just `unset`.

**Comment by markspolakovs** [comment]
I've done as above, but now I'm facing an interesting issue - the first character I type is invisible, and after that there's a space between the last char and my cursor

![image](https://cloud.githubusercontent.com/assets/2904440/26400806/19431f78-4083-11e7-9da7-8d2eaa0f26d2.png)

**Comment by LtSquigs** [comment]
Im getting the same issue with spacing with Babun and Hyper JS on windows, anyone know what might be up?

**Comment by ChaosZero112** [comment]
This is the first hit that comes up when searching Hyper and Babun. Using @schmic and @daagar  suggestions, I have a working setup.

However, I'm having issues with (what assuming is Hyper) autoscrolling down with the content. Once the window fills, it won't continue until I press a key or scroll up then back down.

**Comment by chabou** [comment]
@ChaosZero112 I have no idea but giving your Hyper version could maybe help.

**Comment by ChaosZero112** [comment]
Latest Stable, Hyper 1.4.2. And Babun 1.2.0.
Settings: https://gist.github.com/ChaosZero112/b7cb4573150bb69205b1bae4eeefa436

**Comment by Stanzilla** [comment]
Hey there, please try again with the new Hyper v2 release and open a new issue if the problem stills exists. Thank you!
